Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/WatchdogService.kt

```diff
@@ -0,0 +1,76 @@
+package de.rki.coronawarnapp.util
+
+import android.content.Context
+import android.net.wifi.WifiManager
+import android.os.PowerManager
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.transaction.RetrieveDiagnosisKeysTransaction
+import de.rki.coronawarnapp.worker.BackgroundWorkHelper
+import de.rki.coronawarnapp.worker.BackgroundWorkScheduler
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class WatchdogService @Inject constructor(private val context: Context) {
```

Comment:
Are we sure this does not introduce additional problems based on lifecycle processing? Recommendation from Google is to leave the Logic directly in onCreate without Overhead.

Reply:
> additional problems based on lifecycle processing

Can you elaborate on that? Instantiation via Dagger does not switch threads and does not happen asynchronously.

What dagger does here would be like

```kotlin
    private lateinit var watchdogService: WatchdogService
    override fun onCreate() {
        instance = this
        super.onCreate()

        (...)
        watchdogService = Watchdog(this)

        (...)
        watchdogService.launch()
    }
```

The overhead via dagger will only cost us some nano seconds on the main thread, which I think is ok to pay to gain better abstraction. I don't think we are in an area of delay here that can negatively impact us.

Reply:
While I agree this SHOULD not impose side-effects, we still have some uncertainty when it comes to us being called by the watch-dog. Let's run this through our test-track and see if we have a behaviour change. We should definitely include some logging for the testers though to make sure the launch happens without issues.

Reply:
I've added additional log statements to make that visible.